libraries/stdlib/genglsl/stdlib_genglsl_impl.mtlx

```diff
@@ -6,21 +6,16 @@
 
     Declarations for glsl implementations of standard nodes included in the MaterialX specification.
   -->
-
   <!-- ======================================================================== -->
   <!-- Shader nodes                                                             -->
   <!-- ======================================================================== -->
-
   <!-- <surfacematerial> -->
   <implementation name="IM_surfacematerial_genglsl" nodedef="ND_surfacematerial" target="genglsl" />
-
   <!-- <surface_unlit> -->
   <implementation name="IM_surface_unlit_genglsl" nodedef="ND_surface_unlit" target="genglsl" />
-
```

Comment:
This change looks really promising, thanks @kwokcb!

One question that comes to mind is whether the newlines in these original documents provided any valuable clarity, and if so, whether there's a practical way to maintain them when we apply automated updates to our data libraries over time.

Reply:
I'm not sure how many folks beyond maintainers would read these file. There are section comments already, just not spacing. I did turn on "preserve comments" but these are not comment lines unfortunately. Maybe we could try to add additional lines to existing comments?

Reply:
@kwokcb My sense is that more experienced MaterialX developers do read through this file to improve their understanding of the nodes and their implementations, and I often send out links to individual lines in MaterialX discussions.  I don't yet have a solution in mind for preserving newlines in addition to comments, but it seems worthwhile to consider whether it's practically possible before we remove them from the documents.

Reply:
Fair enough and can either manually restore blank lines, or I think better to insert empty comment lines in the interim as they do get preserved.
It will look like this using comments. 
```xml
<!---->
  <!-- <crossproduct> -->
  <implementation name="IM_crossproduct_vector3_genmsl" nodedef="ND_crossproduct_vector3" target="genmsl" sourcecode="cross({{in1}}, {{in2}})" />
  <!---->
  <!-- <transformpoint> -->
  <implementation name="IM_transformpoint_vector3_genmsl" nodedef="ND_transformpoint_vector3" target="genmsl" />
  <!---->
  <!-- <transformvector> -->
  <implementation name="IM_transformvector_vector3_genmsl" nodedef="ND_transformvector_vector3" target="genmsl" />
  <!---->
  <!-- <transformnormal> -->
  <implementation name="IM_transformnormal_vector3_genmsl" nodedef="ND_transformnormal_vector3" target="genmsl" />
  <!---->
```
Whether support for blank lines can be done can be investigated separately.

That said I'd hope that there be better formal documentation than reading through these files. I did extract out this information to create Markdown/html which can be formatted as desired as part of `mxdoc` script improvements. Then folks can reference docs vs source files.
e.g. This is what I have generated now for code impls for e.g.:
![image](https://user-images.githubusercontent.com/49369885/230964290-db562141-ad1a-4621-833e-0b1307d32433.png)

Reply:
That's an interesting idea, using empty comments instead of newlines, and I'd be open to that approach if it maintains readability.

With respect to more formally documenting node interfaces in HTML, that's definitely useful, but it's worthwhile to make the analogy with C++ source code, where the ground truth always remains the original source code on GitHub, and the C++ API documentation in HTML doesn't remove the need for this source code to remain clear and readable.

Reply:
True enough, but I'd rather read Doxygen docs than C++ code :). 
I could not find any way to parse blank lines as white space is consumed and no DOM nodes are created. I will add in empty comments which do turn into DOM nodes so will work.

Reply:
Agreed, I really like Doxygen-generated API documentation as well, but it's not quite good enough that I'd want to compromise on newlines and whitespace in the C++ source!

Reply:
Thanks for putting together this version with empty comments instead of newlines, and it's definitely an interesting proposal, though I'm not sure this creates a very readable experience for users.

My own, very subjective ranking of legibility would be:

1. Separating sections by newlines (the original version)
2. No separators between sections
3. Separating sections by empty comments (the latest proposal)

What are your thoughts?

Reply:
I do like that idea, but wouldn't want to do it here for risking destabilizing anything. I "think " we just need to look for
lines with only `<eol>` in them and create an new node type which is empty_line node. Then insert it as a child Element like comments.

We can go with 1, but I have one other variant which is split comments across multiple lines to create spacing.
So if you want spacing, add it inside the comment for now. Still a bit ugly.
 <!--
    ======================================================================== 
    Conditional nodes                                                        
    ======================================================================== 
  -->
  <!-- 
    <ifgreater>
  -->

Reply:
That's an interesting idea as well, though I suspect we'll get the finest control over layout using actual newlines, and our current formatting of library documents strikes a pretty good balance between readability and efficient parsing.  If we do add support for newline elements in some form, we would naturally leave this feature disabled by default, to maintain the efficiency of document parsing in existing tools.

Reply:
Sounds good. I can leave an new issue for an empty line / white-space option with some info that I have so far, and will use option 1.